models/balancer/ethereum/balancer_ethereum_vebal_balances_day.sql

```diff
@@ -0,0 +1,144 @@
+{{
+    config(
+        schema = 'balancer_ethereum',
+        alias='vebal_balances_day',
+        post_hook='{{ expose_spells(\'["ethereum"]\',
+                                    "project",
+                                    "balancer",
+                                    \'["markusbkoch", "mendesfabio", "stefenon"]\') }}'
+    )
+}}
+
+WITH base_locks AS (
+        SELECT d.provider, ts AS locked_at, locktime AS unlocked_at, ts AS updated_at
+        FROM {{ source('balancer_ethereum', 'veBAL_call_create_lock') }} l  
+        JOIN {{ source('balancer_ethereum', 'veBAL_evt_Deposit') }} d
+        ON d.evt_tx_hash = l.call_tx_hash
+        
+        UNION ALL
+        
+        SELECT provider, null::numeric AS locked_at, locktime AS unlocked_at, ts AS updated_at
+        FROM {{ source('balancer_ethereum', 'veBAL_evt_Deposit') }}
+        WHERE value = 0
```

Comment:
Hi @Hosuke, this model requires the data to be as fresh as possible. If we could make the incremental view refreshing data that is older than 1 hour, I believe we can do it. Otherwise, it would have a significant impact in the results.

Reply:
Ok, we can see if this model can be a table in the later review. I am not sure the source tables have <1h freshness.

Meanwhile we need to avoid using `::` in the spellbook. We can try using equivalent cast().
```suggestion
        FROM {{ source('balancer_ethereum', 'veBAL_call_create_lock') }} l  
        JOIN {{ source('balancer_ethereum', 'veBAL_evt_Deposit') }} d
        ON d.evt_tx_hash = l.call_tx_hash
        
        UNION ALL
        
        SELECT provider, cast(null as numeric(38)) AS locked_at, locktime AS unlocked_at, ts AS updated_at
        FROM {{ source('balancer_ethereum', 'veBAL_evt_Deposit') }}
        WHERE value = 0
```